qiskit_machine_learning/utils/loss_functions/loss_functions.py

```diff
@@ -13,46 +13,95 @@
 """ Loss utilities """
 
 from abc import ABC, abstractmethod
+from typing import Union, Tuple
 import numpy as np
 
+
 from ...exceptions import QiskitMachineLearningError
 
 
 class Loss(ABC):
     """
-    Abstract base class for Loss.
+    Abstract base class for computing Loss.
+
     """
 
-    def __call__(self, predict, target):
+    def __call__(self, predict: Union[int, np.ndarray], target: Union[int, np.ndarray]) -> float:
+        """
+        Args:
+            predict: a numpy array of predicted values using the model
+            target: a numpy array of the true values
+
+        Returns:
+            a float value of the loss function
+        """
         return self.evaluate(predict, target)
 
     @abstractmethod
-    def evaluate(self, predict, target):
-        """evaluate"""
+    def evaluate(self, predict: Union[int, np.ndarray], target: Union[int, np.ndarray]) -> float:
+        """
+        An abstract method for evaluating the loss function
+
+        Args:
+            predict: a numpy array of predicted values using the model
+            target: a numpy array of the true values
+
+        Raises:
+            QiskitMachineLearningError: shapes of predict and target do not match
+        """
         raise NotImplementedError
 
     @abstractmethod
-    def gradient(self, predict, target):
-        """gradient"""
+    def gradient(self, predict: Union[int, np.ndarray], target: Union[int, np.ndarray]) -> float:
+        """
+        An abstract method for computing the gradient
+
+        Args:
+            predict: a numpy array of predicted values using the model
+            target: a numpy array of the true values
+
+        Raises:
+            QiskitMachineLearningError: shapes of predict and target do not match.
+        """
         raise NotImplementedError
 
     @staticmethod
-    def _validate(predict, target):
+    def _validate(
+        predict: Union[int, np.ndarray], target: Union[int, np.ndarray]
+    ) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
So if I pass in an 'int' and an 'int' as predict and target, I will not get back an Tuple of ints rather a Tuple of arrays? I guess its doing a type cast as well as validate that the other methods rely on. Not sure if it would be helpful to expose the duality in the name say `_cast_validate` or `_to_array_validate` - its a private method so maybe its less important.

Reply:
> 
> 
> So if I pass in an 'int' and an 'int' as predict and target, I will not get back an Tuple of ints rather a Tuple of arrays? I guess its doing a type cast as well as validate that the other methods rely on. Not sure if it would be helpful to expose the duality in the name say `_cast_validate` or `_to_array_validate` - its a private method so maybe its less important.

Now only numpy arrays, renamed to `_validate_shapes`. I hope it is good enough now.